### DISCUSSION TITLE
"Warning: musl detected, skipping GLIBC check" in devcontainers.json?

### DISCUSSION THREAD

**Discussion by dekiesel** [devcontainers/community#110] [Q&A]
After adding the `debugpy` extension  to the container I am seeing this line in the container config, is that to be expected?

image: `python3.12-alpine3.19`
vscode: `1.86`

![image](https://github.com/devcontainers/community/assets/17333827/a0794345-21dc-49b2-9f47-0fdf2e6c777c)

Apologies if this is not the right discussion board.

**Comment by samruddhikhandale** [comment]
Hi 👋 

The warning message "Warning: musl detected, skipping GLIBC check" is indicating that your container is using the `musl` library as its C library instead of `glibc`.

The `debugpy` extension is checking for `glibc` because it has some dependencies on it. However, when it detects that `musl` is being used instead, it skips the `glibc` check and issues a warning. This doesn't necessarily mean that `debugpy` won't work, but there might be some features that don't behave as expected.

I was able to provide a generic information on who and why adds the warning that you are seeing. If you need more help, I'd recommend reaching out to the maintainers of the `debugpy` extension. Thanks!

**Reply by dekiesel** [reply]
Hi and thank you for your answer. I am surprised to see this output in the container config. Imo it shouldn't be in the `extensions` section of the container config at all. Could you confirm that?

**Reply by samruddhikhandale** [reply]
Unfortunately, I am not familiar with `debugpy` extension, hence, it would be great to contact it's maintainers.

>  Imo it shouldn't be in the extensions section of the container config at all. Could you confirm that?

Also, can you open an issue in https://github.com/microsoft/vscode-remote-release/issues? The VS Code team should be able to help you answer this question, thanks!
